assignment/assignment-tool/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java

```diff
@@ -7418,11 +7426,25 @@ else if (!Integer.valueOf(Assignment.NON_ELECTRONIC_ASSIGNMENT_SUBMISSION).equal
 					}
 					else
 					{
-						validPointGrade(state, gradePoints);
+						Integer scaleFactor = AssignmentService.getScaleFactor();
+						try {
+							if (StringUtils.isNotEmpty(assignmentRef)) {
+								Assignment assignment = AssignmentService.getAssignment(assignmentRef);
+								if (assignment != null && assignment.getContent() != null) {
+									scaleFactor = assignment.getContent().getFactor();
+								}
+							}
+						} catch (IdUnusedException e) {
+							M_log.warn(e);
+						} catch (PermissionException e) {
+							M_log.warn(e);
+						}
```

Comment:
Since you're doing the same thing for both exception types, you can use a multi catch:

catch (IdUnusedException | PermissionException e) { 
    M_log.warn(e);
}